### ISSUE TITLE
update to slinky 0.6.4 / scala-js 1.0

### ISSUE THREAD

**Issue by tyrcho** [shadaj/create-react-scala-app.g8#10]
Hi,

would you please update the template ? I've tried manually but I get this error:

```
[IJ]sbt:slinky-demo> dev
[info] Compiling 8 Scala sources to C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\classes ...
[info] Non-compiled module 'compiler-bridge_2.13' for Scala 2.13.1. Compiling...
[info] Updating NPM dependencies
8 warnings found
[info]   Compilation completed in 16.42s.
[info] Done compiling.
[info] Fast optimizing C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\slinky-demo-fastopt.js
[info] > webpack-cli@3.3.2 postinstall C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\node_modules\webpack-cli
[info] > node ./bin/opencollective.js


[info]                             Thanks for using Webpack!
[info]                  Please consider donating to our Open Collective


[info]                         to help us maintain this package.
[info]                  Donate: https://opencollective.com/webpack/donate
[error] npm notice created a lockfile as package-lock.json. You should commit this file.
[error] npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules\chokidar\node_modules\fsevents):
[error] npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.12: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
[error] npm WARN optional SKIPPING OPTIONAL DEPENDENCY: abbrev@1.1.1 (node_modules\fsevents\node_modules\abbrev):
[error] npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename 'C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\node_modules\fsevents\node_modules\abbrev' -> 'C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\node_modules\fsevents\node_modules\.abbrev.DELETE'
[error] npm WARN optional SKIPPING OPTIONAL DEPENDENCY: ansi-regex@2.1.1 (node_modules\fsevents\node_modules\ansi-regex):
[error] npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename 'C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\node_modules\fsevents\node_modules\ansi-regex' -> 'C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\node_modules\fsevents\node_modules\.ansi-regex.DELETE'


[error] npm WARN optional SKIPPING OPTIONAL DEPENDENCY: aproba@1.2.0 (node_modules\fsevents\node_modules\aproba):

...

[error] npm WARN optional SKIPPING OPTIONAL DEPENDENCY: yallist@3.1.1 (node_modules\fsevents\node_modules\yallist):
[error] npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename 'C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\node_modules\fsevents\node_modules\yallist' -> 'C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\node_modules\fsevents\node_modules\.yallist.DELETE'
[info] Writing module entry point for slinky-demo-fastopt-entrypoint.js
[info] Building webpack library bundles for slinky-demo-fastopt in C:\Users\tyrcho\git\slinky-demo\target\streams\_global\_global\_global\streams\fastOptJS-webpack-libraries
[info] Writing scalajs.webpack.config.js
[info] Version: 4.29.6
[info] Hash: 90e92abfb87706d71c30
[info] Time: 1563ms
[info] Path: C:\Users\tyrcho\git\slinky-demo\target\scala-2.13\scalajs-bundler\main\dist
[info] Built at 2020-04-04T11:34:18.951318800
[info]                  Asset      Size                              Chunks
[info]            favicon.ico      3870   [emitted]                      []
[info]             index.html      1532   [emitted]                      []
[info]          manifest.json       332   [emitted]                      []
[info] slinky-demo-fastopt.js   4056808   [emitted]   [slinky-demo-fastopt]
[info] Starting webpack-dev-server
[success] Total time: 55 s, completed 4 avr. 2020 ├á 11:34:19
[success] Total time: 0 s, completed 4 avr. 2020 ├á 11:34:20
[info] 1. Monitoring source files for slinky-demo/fastOptJS...
[info]    Press <enter> to interrupt or '?' for more options.
[error] internal/modules/cjs/loader.js:596
[error]     throw err;
[error]     ^
[error] Error: Cannot find module 'webpack'
[error]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
[error]     at Function.Module._load (internal/modules/cjs/loader.js:520:25)
[error]     at Module.require (internal/modules/cjs/loader.js:650:17)
[error]     at require (internal/modules/cjs/helpers.js:20:18)
[error]     at Object.<anonymous> (C:\Users\tyrcho\node_modules\webpack-cli\bin\convert-argv.js:7:24)
[error]     at Module._compile (internal/modules/cjs/loader.js:702:30)
[error]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
[error]     at Module.load (internal/modules/cjs/loader.js:612:32)
[error]     at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
[error]     at Function.Module._load (internal/modules/cjs/loader.js:543:3)
```

**Comment by tyrcho** [comment]
Here is my `build.sbt`

```scala
enablePlugins(ScalaJSBundlerPlugin)

name := "slinky-demo"

scalaVersion := "2.13.1"

Global / onChangedBuildSource := ReloadOnSourceChanges

npmDependencies in Compile += "react" -> "16.12.0"
npmDependencies in Compile += "react-dom" -> "16.12.0"
npmDependencies in Compile += "react-proxy" -> "1.1.8"

npmDevDependencies in Compile += "file-loader" -> "3.0.1"
npmDevDependencies in Compile += "style-loader" -> "0.23.1"
npmDevDependencies in Compile += "css-loader" -> "2.1.1"
npmDevDependencies in Compile += "html-webpack-plugin" -> "3.2.0"
npmDevDependencies in Compile += "copy-webpack-plugin" -> "5.0.2"
npmDevDependencies in Compile += "webpack-merge" -> "4.2.1"

libraryDependencies += "me.shadaj" %%% "slinky-web" % "0.6.4"
libraryDependencies += "me.shadaj" %%% "slinky-hot" % "0.6.4"

libraryDependencies += "org.scalatest" %%% "scalatest" % "3.1.1" % Test

scalacOptions += "-P:scalajs:sjsDefinedByDefault"
scalacOptions += "-Ymacro-annotations"

version in webpack := "4.29.6"
version in startWebpackDevServer:= "3.2.1"

webpackResources := baseDirectory.value / "webpack" * "*"

webpackConfigFile in fastOptJS := Some(baseDirectory.value / "webpack" / "webpack-fastopt.config.js")
webpackConfigFile in fullOptJS := Some(baseDirectory.value / "webpack" / "webpack-opt.config.js")
webpackConfigFile in Test := Some(baseDirectory.value / "webpack" / "webpack-core.config.js")

webpackDevServerExtraArgs in fastOptJS := Seq("--inline", "--hot")
webpackBundlingMode in fastOptJS := BundlingMode.LibraryOnly()

requireJsDomEnv in Test := true

addCommandAlias("dev", ";fastOptJS::startWebpackDevServer;~fastOptJS")

addCommandAlias("build", "fullOptJS::webpack")

```

**Comment by nicflores** [comment]
@tyrcho, this `build.sbt` file allowed me to run `sbt dev`. Hopefully this will help you. If I have time I will PR this change this weekend if no one else gets to it before then.

```scala
enablePlugins(ScalaJSBundlerPlugin)

name := "onestopcalc-slinky"

scalaVersion := "2.13.2"

Global / onChangedBuildSource := ReloadOnSourceChanges

npmDependencies in Compile += "react" -> "16.13.1"
npmDependencies in Compile += "react-dom" -> "16.13.1"
npmDependencies in Compile += "react-proxy" -> "1.1.8"

npmDevDependencies in Compile += "file-loader" -> "6.0.0"
npmDevDependencies in Compile += "style-loader" -> "1.2.1"
npmDevDependencies in Compile += "css-loader" -> "3.5.3"
npmDevDependencies in Compile += "html-webpack-plugin" -> "4.3.0"
npmDevDependencies in Compile += "copy-webpack-plugin" -> "5.1.1"
npmDevDependencies in Compile += "webpack-merge" -> "4.2.1"

libraryDependencies += "me.shadaj" %%% "slinky-core" % "0.6.5" // core React functionality, no React DOM
libraryDependencies += "me.shadaj" %%% "slinky-web" % "0.6.5" // React DOM, HTML and SVG tags
libraryDependencies += "me.shadaj" %%% "slinky-native" % "0.6.5" // React Native components
libraryDependencies += "me.shadaj" %%% "slinky-hot" % "0.6.5" // Hot loading, requires react-proxy package
//libraryDependencies += "me.shadaj" %%% "slinky-scalajsreact-interop" % "0.6.5" // Interop with japgolly/scalajs-react
libraryDependencies += "org.scalatest" %%% "scalatest" % "3.1.1" % Test

//scalacOptions += "-P:scalajs:sjsDefinedByDefault"
//addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.1" cross CrossVersion.full)

// if using Scala 2.13.0, instead use
scalacOptions += "-Ymacro-annotations"

version in webpack := "4.43.0"
version in startWebpackDevServer:= "3.11.0"

webpackResources := baseDirectory.value / "webpack" * "*"

webpackConfigFile in fastOptJS := Some(baseDirectory.value / "webpack" / "webpack-fastopt.config.js")
webpackConfigFile in fullOptJS := Some(baseDirectory.value / "webpack" / "webpack-opt.config.js")
webpackConfigFile in Test := Some(baseDirectory.value / "webpack" / "webpack-core.config.js")

webpackDevServerExtraArgs in fastOptJS := Seq("--inline", "--hot")
webpackBundlingMode in fastOptJS := BundlingMode.LibraryOnly()

requireJsDomEnv in Test := true

addCommandAlias("dev", ";fastOptJS::startWebpackDevServer;~fastOptJS")

addCommandAlias("build", "fullOptJS::webpack")
```

**Comment by shadaj** [comment]
Resolved by #12!
